cli: don't run update-notifier in CI environment

sibiraj-s opened the pull request:
This Prevents update notifier from running in CI environment

sibiraj-s committed:
cli: don't run update-notifier in CI environment

sibiraj-s committed:
chore: update bundleDependencies

olore commented:
Nice one @Sibiraj-S. This and #32 will make the update-notifier work the same as pnpm

zkat commented:
Ahh, I commented on #32 before seeing this PR. @olore feel free to ignore my comments related to CI itself.

zkat commented on bin/npm-cli.js:
```diff
@@ -80,9 +80,10 @@
     ) {
       const pkg = require('../package.json')
       let notifier = require('update-notifier')({pkg})
+      const isCI = require('is-ci')
```
I'd rather use [`ci-info`](https://npm.im/ci-info), since it's what `is-ci` itself uses. We don't need yet another package.json floating around for a one-liner by the same author. `ci-info` exports an `isCI` property that you can achieve the same thing with.

sibiraj-s replied:
@zkat . It seems `is-ci` is already a part of this package(a dependant of update-notifier). having that now won't make a difference?

zkat commented on bin/npm-cli.js:
```diff
@@ -80,9 +80,10 @@
       if (
         notifier.update &&
-        notifier.update.latest !== pkg.version
+        notifier.update.latest !== pkg.version && !isCI
```
do you mind putting this on the next line for consistency?

zkat commented on package.json:
```diff
@@ -177,6 +178,7 @@
     "inherits",
     "ini",
     "init-package-json",
+    "is-ci",
```
Awesome! You added it to `bundleDependencies`. Additionally, we commit our dependencies, and we use `package-lock.json` so please go ahead and `git add -A node_modules/ package-lock.json` as well before you commit the `ci-info` change I asked for above!

zkat reviewed (changes requested):
So far so good! Thanks a bunch for submitting this PR, and welcome! Just a couple of changes and I think this is ready to go. 🎉

Sibiraj-S commented:
@zkat, all changes are now addressed. At first I didn't quite understand the behaviour of lock file in this repo, that why I skipped it purposely last time. I have added it now.

iarna commented on bin/npm-cli.js:
```diff
@@ -80,9 +80,11 @@
     ) {
       const pkg = require('../package.json')
       let notifier = require('update-notifier')({pkg})
+      const isCI = require('ci-info').isCI
```
We already detect CI in a handful of places and currently `ci-info` doesn't entirely align. Then again, our various detectors don't align with each other. Ideally, I'd like to see those patched to use `ci-info` as well (but not in this PR).

For reference, our existing CI detectors are here:

https://github.com/npm/cli/blob/ab3c62aa83375a7d4a21e396129ae5679ee86646/lib/config/defaults.js#L194

and here:

https://github.com/npm/npm-registry-fetch/blob/03dde52817d00996d8316b155832d8802f338236/config.js#L26

The latter one detects things `ci-info` doesn't, so `ci-info` will need patching before we can entirely switch over to it.

There's also:

https://github.com/npm/npm-registry-client/blob/d70399bff938b04cd3ed7052f415c6ed2d6bd275/lib/initialize.js#L15

Which should probably be patched too, though we aren't/won't be using it any more in npm proper.

Sibiraj-S replied:
I that case. @iarna . `ci-info` needs a PR I guess. or I can replace `ci-info` with existing one. which one would be better?

watson replied:
BTW: The `is-ci` module is just a convenience wrapper around `ci-info`, that when used programmatically returns a boolean and when used from the command line uses the exit code to signal if it's being run on a CI server or not.

iarna commented on package.json:
```diff
@@ -45,6 +45,7 @@
     "cacache": "^11.0.2",
     "call-limit": "~1.1.0",
     "chownr": "~1.0.1",
+    "ci-info": "^1.1.3",
```
There's still no `package-lock.json` change in this PR? We'll need that to land this.

Sibiraj-S replied:
I remember commiting that lock file `node_modules/ci-info/package.json`. isn't that the one?

sibiraj-s replied:
@iarna I got it. The `ci-info` is already used as a dependecy of `update-notifier`

`update-notifier -> is-ci -> ci-info`

zkat replied:
ohhhhh

that makes sense.

sibiraj-s commented:
@iarna . I raised a PR in `ci-info` for detecting `solano-ci`. which is the only ci which is not detected by ci-info and used here.

https://github.com/watson/ci-info/pull/9

Sibiraj-S commented:
@zkat . Anything else to update?

zkat commented:
Once @watson gets around to reviewing/approving the `ci-info` changes, I think we'll be ready to merge this. So I figure I'll ping him 😅

zkat commented:
As a note for @iarna: I think it's fine and pretty much safe that `ci-info` uses different env var keys for Bamboo. If one is present, the other one is also present. And if you're being _that_ clever about tricking npm into thinking you're in bamboo, then I dunno what to tell ya. I think it's fine to accept this once `ci-info` gets the solano patch. Everything else _seems_ to be the same.

Additional note to self: I should probably patch `npm-registry-fetch` to use this as well. Maybe `pacote` if this gets merged before we upgrade pacote to the n-r-f version.

watson commented:
@zkat Thanks a lot of pinging me on this. I'm so bad at checking github notifications - have a huge backlog 😬

sibiraj-s committed:
perf: use `ci-info` instead of `is-ci`

sibiraj-s commented:
`ci-info` updated to v1.3.1. detects `solano-ci`

sibiraj-s committed:
chore: bump ci-info to v1.3.1

zkat reviewed (approved):
👍 looks good! Thank you @watson and @Sibiraj-S!

zkat commented:
I merged this manually, so github isn't gonna detect it as a merge, but this is now in `release-next`: https://github.com/npm/cli/commit/cabc0f07a9102a769eeb54de2b2c6b0c1a979b35

sibiraj-s commented:
cool. @zkat I would also suggest to update ci-info to v1.4 before the release, as it detects two more popular CI's
